test/e2e/agent_manager.go

```diff
@@ -0,0 +1,283 @@
+	}
+
+	// Start the process
+	if err := m.cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start agent process: %w", err)
+	}
+
+	// Stream logs in background (filter out noise)
+	go m.streamLogs(stdout, "STDOUT")
+	go m.streamLogs(stderr, "STDERR")
+
+	// Wait a bit for agent to start
+	time.Sleep(2 * time.Second)
+
+	// Check if process is still running
+	if m.cmd.Process == nil {
+		return fmt.Errorf("agent process failed to start")
+	}
+
+	return nil
+}
+
+// streamLogs streams logs from the agent
+func (m *AgentManager) streamLogs(reader io.Reader, prefix string) {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		
+		// Filter to only show important agent activity
+		// Show: startup, probe processing, status updates, errors
+		// Skip: routine reconciliation cycles, fetching counts, prometheus messages
+		if containsAny(line, []string{
+			"Agent started with config",
+			"Processing probe",
+			"Successfully updated probe",
+			"Successfully processed probe",
+			"Failed to create Kubernetes",
+			"error", "ERROR", "Error:",
+			"Probe .* processed",
+		}) {
+			// Skip overly verbose source info in JSON logs
+			if strings.Contains(line, `"source":`) {
+				// Parse JSON and extract just time, level, and msg
+				var logEntry map[string]interface{}
+				if err := json.Unmarshal([]byte(line), &logEntry); err == nil {
+					if msg, ok := logEntry["msg"].(string); ok {
+						level := "INFO"
+						if l, ok := logEntry["level"].(string); ok {
+							level = l
+						}
+						fmt.Printf("[Agent %s] %s: %s\n", prefix, level, msg)
+						continue
+					}
+				}
+			}
+			fmt.Printf("[Agent %s] %s\n", prefix, line)
+		}
+	}
+}
+
+// containsAny checks if a string contains any of the given substrings
+func containsAny(s string, substrs []string) bool {
+	for _, substr := range substrs {
+		if strings.Contains(s, substr) {
+			return true
+		}
+	}
+	return false
+}
+
+// Stop stops the agent
+func (m *AgentManager) Stop() error {
+	if !m.started {
+		return nil
+	}
+
+	// Cancel context to signal shutdown
+	m.cancel()
+
+	// Send SIGTERM to process group
+	if m.cmd != nil && m.cmd.Process != nil {
+		pgid, err := syscall.Getpgid(m.cmd.Process.Pid)
+		if err == nil {
+			_ = syscall.Kill(-pgid, syscall.SIGTERM)
+		}
+
+		// Wait for process to exit with timeout
+		done := make(chan error, 1)
+		go func() {
+			done <- m.cmd.Wait()
+		}()
+
+		select {
+		case <-done:
+			// Process exited
+		case <-time.After(5 * time.Second):
+			// Force kill if still running
+			if m.cmd.Process != nil {
+				_ = m.cmd.Process.Kill()
+			}
+		}
+	}
+
+	// Clean up config file
+	if m.configPath != "" {
+		_ = os.Remove(m.configPath)
+	}
+
+	m.started = false
+	close(m.stopChan)
+
+	return nil
+}
+
+// isPortAvailable checks if a port is available
+func isPortAvailable(port int) bool {
+	address := fmt.Sprintf(":%d", port)
+	listener, err := net.Listen("tcp", address)
```

Comment:
> For TCP networks, if the host in the address parameter is empty or a literal unspecified IP address, Listen listens on all available unicast and anycast IP addresses of the local system.

This opens that port on the host to the entire network.  I think we should default to localhost/127.0.0.1 unless we're _explicitly_ clear to anyone running this test that they're opening a pubic port.